nodejs/week2/app.js

```diff
@@ -0,0 +1,89 @@
+import express from "express";
+import fs from "fs/promises";
+import path from "path";
+import { fileURLToPath } from "url";
+
+const app = express();
+const port = process.env.PORT || 3000;
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const filePath = path.join(__dirname, "documents.json");
+
+app.use(express.json());
+
+app.get("/search", async (req, res) => {
+  try {
+    const data = await fs.readFile(filePath, "utf-8");
+    const documents = JSON.parse(data);
+    if (req.query.q) {
+      const searchString = req.query.q.toLowerCase();
+      const filteredDocuments = documents.filter((document) =>
+        Object.values(document)
+          .map((ele) => String(ele).toLowerCase())
+          .some((value) => value.includes(searchString))
+      );
+      return res.json(filteredDocuments);
+    } else {
+      return res.json(documents);
+    }
+  } catch (error) {
+    res.status(404).json({ message: "Error parsing" });
+  }
+});
+
+app.post("/search", async (req, res) => {
+  try {
+    const q = req.query.q;
+    const field = req.body.fields;
+
+    if (q && field) {
+      return res
+        .status(404)
+        .json({ message: "Cannot provide both the values" });
+    }
+    const data = await fs.readFile(filePath, "utf-8");
+    const documents = JSON.parse(data);
+    let filteredDocuments = documents;
+
+    if (q) {
+      filteredDocuments = filteredDocuments.filter((document) =>
+        Object.values(document).some((value) =>
+          String(value).toLowerCase().includes(q.toLowerCase())
+        )
+      );
+    }
+    if (field) {
+      for (const property in field) {
+        filteredDocuments = filteredDocuments.filter(
+          (document) =>
+            document[property] && String(document[property]) == field[property]
+        );
+      }
+    }
+    return res.json(filteredDocuments);
+  } catch (error) {
+    return res.status(404).json({ message: "Error parsing" });
```

Comment:
In case of server error we return 500-599 codes [Server error responses](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses)

```suggestion
    return res.status(500).json({ message: "Error parsing" });
```